DEVELOPER.md

```diff
@@ -0,0 +1,374 @@
+
+### [Cli](cli)
+
+Cli is an example command line interface that uses the [TestManager](model/src/test_manager/manager.rs) to interact with a TestSys cluster.
+
+### [Controller](controller)
+
+The controller orchestrates the interaction between TestSys tests and resources. 
+After a TestSys Test CRD or TestSys Resource CRD has been added to the cluster (`cli run file`), the controller begins the reconciliation process.
+
+For tests, once necessary resources have been created (resources named in the `resources` field of the test spec), a k8s job is created and the test agent is run.
+
+For resources, once necessary resources have been created (resources named in the `depends_on` field of the resource spec), a k8s job is created and the `create` function of the agent is run.
+Once a resource is marked for deletion (`cli delete`), the `destroy` function of the resource is run, and the finalizers are removed so that the resource can be cleaned up by k8s.
+
+For more info, see the [design](design/DESIGN.md) document. 
+
+### [Model](model)
+
+The model library contains the infrastructure that TestSys is built upon.
+This includes the CRDs for [tests](model/src/test.rs) and [resources](model/src/resource.rs) as well as [everything](model/system) needed to be installed to the TestSys cluster.
+
+All interactions with the TestSys cluster should be done using the model library through `TestManager`, `TestClient`, or `ResourceClient`.
+
+The model provides [TestManager](model/src/test_manager/manager.rs) which can be used as an entry point for user interactions with TestSys.
+To see how to use the TestManager look at the [example cli](cli).
+
+Model also contains a [TestClient](model/src/clients/test_client) for simple interactions with Tests and a [ResourceClient](model/src/clients/resource_client) for Resources.
+
+### [Selftest](selftest)
+
+Selftest is a library used to validate incoming pull request by testing the overall TestSys framework.
+
+## Getting Started
+
+TestSys is currently tested on `x86_64` machines, these steps may not work as expected on other operating systems.
+
+### Setting up TestSys
+
+#### Create a TestSys directory
+Create a directory that you wish use as a working directory and navigate to it. 
+The example provided creates a directory in the users home, but a directory may be created anywhere.
+
+```bash
+cd ~
+mkdir testsys && cd testsys
+pwd 
+```
+
+#### Clone the TestSys repo
+While in the TestSys directory, clone the `bottlerocket-test-system` repo from github.
+
+```bash
+git clone https://github.com/bottlerocket-os/bottlerocket-test-system.git
+cd bottlerocket-test-system
+export TESTSYS_DIR=$(pwd)
+```
+
+The TestSys directory is now set up.
+
+### Installing tools
+
+#### Install `Kind`
+Instead of creating a long-standing eks cluster for TestSys development, `kind` enables the quick creation and destruction of local clusters that exist within the developers Docker which helps improve the testing experience.
```

Comment:
nit:
```suggestion
Instead of creating a long-standing EKS cluster for TestSys development, `kind` enables the quick creation and destruction of local clusters that exist within the developers Docker which helps improve the testing experience.
```

and potentially elsewhere